Index: src/Makefile
===================================================================
--- src/Makefile	(revision 901)
+++ src/Makefile	(working copy)
@@ -4,8 +4,10 @@
   TECMAKE_CMD = $(MAKE) --no-print-directory -f ../tecmake.mak
 endif
 
-.PHONY: do_all cd cdpdf cdgl cdim cdcontextplus cdlua5 cdluapdf5 cdluagl5 cdluacontextplus5 cdluaim5
-do_all:        cd cdpdf cdgl cdim cdcontextplus cdlua5 cdluapdf5 cdluagl5 cdluacontextplus5 cdluaim5
+TARGETS := cd cdpdf cdgl cdim cdcontextplus cdlua5 cdluapdf5 cdluagl5 cdluacontextplus5 cdluaim5
+TARGETS := $(filter-out $(EXCLUDE_TARGETS), $(TARGETS))
+.PHONY: do_all $(TARGETS)
+do_all: $(TARGETS)
 
 cd:
 	@$(TECMAKE_CMD)
Index: src/cdcontextplus.mak
===================================================================
--- src/cdcontextplus.mak	(revision 901)
+++ src/cdcontextplus.mak	(working copy)
@@ -38,3 +38,4 @@
 
 USE_CD = YES
 CD = ..
+INCLUDES += ../include
Index: src/cdgl.mak
===================================================================
--- src/cdgl.mak	(revision 901)
+++ src/cdgl.mak	(working copy)
@@ -27,3 +27,5 @@
   endif
   LDIR += $(GTK)/lib
 endif
+
+INCLUDES += ../include
Index: src/cdim.mak
===================================================================
--- src/cdim.mak	(revision 901)
+++ src/cdim.mak	(working copy)
@@ -16,3 +16,6 @@
     BUILD_DYLIB=Yes
   endif
 endif
+
+INCLUDES += ../include ../../im/include
+
Index: src/cdlua5.mak
===================================================================
--- src/cdlua5.mak	(revision 901)
+++ src/cdlua5.mak	(working copy)
@@ -48,3 +48,5 @@
 LUAMOD_DIR = Yes
 USE_CD = YES
 CD = ..
+
+INCLUDES += ../include
Index: src/cdluacontextplus5.mak
===================================================================
--- src/cdluacontextplus5.mak	(revision 901)
+++ src/cdluacontextplus5.mak	(working copy)
@@ -67,3 +67,5 @@
   INCLUDES += ../include
   LDIR = ../lib/$(TEC_UNAME)
 endif
+
+INCLUDES += ../include
Index: src/cdluagl5.mak
===================================================================
--- src/cdluagl5.mak	(revision 901)
+++ src/cdluagl5.mak	(working copy)
@@ -57,3 +57,5 @@
   INCLUDES += ../include
   LDIR = ../lib/$(TEC_UNAME)
 endif
+
+INCLUDES += ../include ../../im/include
Index: src/cdluaim5.mak
===================================================================
--- src/cdluaim5.mak	(revision 901)
+++ src/cdluaim5.mak	(working copy)
@@ -59,3 +59,5 @@
   INCLUDES += ../include $(IM)/include
   LDIR = ../lib/$(TEC_UNAME) $(IM)/lib/$(TEC_UNAME)
 endif
+
+INCLUDES += ../include ../../im/include
Index: src/cdpdf.mak
===================================================================
--- src/cdpdf.mak	(revision 901)
+++ src/cdpdf.mak	(working copy)
@@ -8,7 +8,7 @@
 
 INCLUDES = . sim $(PDFLIB)/include
 DEPENDDIR = dep
-LINK_PDFLIB = Yes
+LINK_PDFLIB =
 
 USE_CD = YES
 CD = ..
Index: tecmake.mak
===================================================================
--- tecmake.mak	(revision 901)
+++ tecmake.mak	(working copy)
@@ -251,7 +251,7 @@
 
 #---------------------------------#
 # Compilation Flags
-STDFLAGS := -Wall
+STDFLAGS := -Wall -Wno-incompatible-pointer-types
 STDDEFS  := -DTEC_UNAME=$(TEC_UNAME) -DTEC_SYSNAME=$(TEC_SYSNAME) -D$(TEC_SYSNAME)=$(TEC_SYSRELEASE) -D$(TEC_BYTEORDER) -D$(TEC_WORDSIZE) -DFUNCPROTO=15
 STDINCS  :=
 OPTFLAGS := -O2
