Index: src/Makefile
===================================================================
--- src/Makefile	(revision 901)
+++ src/Makefile	(working copy)
@@ -4,8 +4,10 @@
   TECMAKE_CMD = $(MAKE) --no-print-directory -f ../tecmake.mak
 endif
 
-.PHONY: do_all cd cdpdf cdgl cdim cdcontextplus cdlua5 cdluapdf5 cdluagl5 cdluacontextplus5 cdluaim5
-do_all:        cd cdpdf cdgl cdim cdcontextplus cdlua5 cdluapdf5 cdluagl5 cdluacontextplus5 cdluaim5
+TARGETS := cd cdpdf cdgl cdim cdcontextplus cdlua5 cdluapdf5 cdluagl5 cdluacontextplus5 cdluaim5
+TARGETS := $(filter-out $(EXCLUDE_TARGETS), $(TARGETS))
+.PHONY: do_all $(TARGETS)
+do_all: $(TARGETS)
 
 cd:
 	@$(TECMAKE_CMD)
Index: src/cdpdf.mak
===================================================================
--- src/cdpdf.mak	(revision 901)
+++ src/cdpdf.mak	(working copy)
@@ -8,7 +8,7 @@
 
 INCLUDES = . sim $(PDFLIB)/include
 DEPENDDIR = dep
-LINK_PDFLIB = Yes
+LINK_PDFLIB =
 
 USE_CD = YES
 CD = ..
Index: tecmake.mak
===================================================================
--- tecmake.mak	(revision 901)
+++ tecmake.mak	(working copy)
@@ -251,7 +251,7 @@
 
 #---------------------------------#
 # Compilation Flags
-STDFLAGS := -Wall
+STDFLAGS := -Wall -Wno-incompatible-pointer-types
 STDDEFS  := -DTEC_UNAME=$(TEC_UNAME) -DTEC_SYSNAME=$(TEC_SYSNAME) -D$(TEC_SYSNAME)=$(TEC_SYSRELEASE) -D$(TEC_BYTEORDER) -D$(TEC_WORDSIZE) -DFUNCPROTO=15
 STDINCS  :=
 OPTFLAGS := -O2
