Index: include/cd.h
===================================================================
--- include/cd.h	(revision 901)
+++ include/cd.h	(working copy)
@@ -522,7 +522,10 @@
   CD_LEGAL
 };
 
+// kenjirofukuda@gmail.com
+typedef void (*VoidArgProc)(cdCanvas* canvas, void *arg);
 
+
 #ifdef __cplusplus
 }
 #endif
Index: src/Makefile
===================================================================
--- src/Makefile	(revision 901)
+++ src/Makefile	(working copy)
@@ -4,14 +4,15 @@
   TECMAKE_CMD = $(MAKE) --no-print-directory -f ../tecmake.mak
 endif
 
-.PHONY: do_all cd cdpdf cdgl cdim cdcontextplus cdlua5 cdluapdf5 cdluagl5 cdluacontextplus5 cdluaim5
-do_all:        cd cdpdf cdgl cdim cdcontextplus cdlua5 cdluapdf5 cdluagl5 cdluacontextplus5 cdluaim5
+.PHONY: do_all cd cdgl cdim cdcontextplus cdlua5  cdluagl5 cdluacontextplus5 cdluaim5
+do_all:        cd cdgl cdim cdcontextplus cdlua5  cdluagl5 cdluacontextplus5 cdluaim5
 
 cd:
 	@$(TECMAKE_CMD)
   
-cdpdf:
-	@$(TECMAKE_CMD) MF=cdpdf
+#cdpdf:
+#	@$(TECMAKE_CMD) MF=cdpdf
+
 cdgl:
 	@$(TECMAKE_CMD) MF=cdgl
 cdim:
Index: src/cdpdf.mak
===================================================================
--- src/cdpdf.mak	(revision 901)
+++ src/cdpdf.mak	(working copy)
@@ -8,7 +8,7 @@
 
 INCLUDES = . sim $(PDFLIB)/include
 DEPENDDIR = dep
-LINK_PDFLIB = Yes
+LINK_PDFLIB = 
 
 USE_CD = YES
 CD = ..
Index: src/config.mak
===================================================================
--- src/config.mak	(revision 901)
+++ src/config.mak	(working copy)
@@ -75,11 +75,13 @@
 USE_FREETYPE = Yes
 USE_ZLIB = Yes
 
+
 ifdef USE_GDK
   USE_GTK = Yes
   CHECK_GTK = Yes
   LIBS = pangocairo-1.0 cairo
   DEFINES += GTK_DISABLE_DEPRECATED 
+  INCLUDES += /usr/include/gtk-3.0 
   SRC += drv/cd0wmf.c
   ifdef USE_GTK3
     DEFINES += GDK_DISABLE_DEPRECATED GSEAL_ENABLE USE_GTK3
Index: tecmake.mak
===================================================================
--- tecmake.mak	(revision 901)
+++ tecmake.mak	(working copy)
@@ -251,7 +251,7 @@
 
 #---------------------------------#
 # Compilation Flags
-STDFLAGS := -Wall
+STDFLAGS := -Wall -Wno-incompatible-pointer-types
 STDDEFS  := -DTEC_UNAME=$(TEC_UNAME) -DTEC_SYSNAME=$(TEC_SYSNAME) -D$(TEC_SYSNAME)=$(TEC_SYSRELEASE) -D$(TEC_BYTEORDER) -D$(TEC_WORDSIZE) -DFUNCPROTO=15
 STDINCS  :=
 OPTFLAGS := -O2
@@ -822,7 +822,7 @@
 LUA53 ?= $(TECTOOLS_HOME)/lua53
 LUA54 ?= $(TECTOOLS_HOME)/lua54
 FTGL  ?= $(TECTOOLS_HOME)/ftgl
-PDFLIB ?= $(TECTOOLS_HOME)/pdflib7
+# PDFLIB ?= $(TECTOOLS_HOME)/pdflib7
 # Freetype and zlib in Linux we use from the system
 
 
@@ -1537,6 +1537,7 @@
   LDIR := $(addprefix -L, $(LDIR))
 endif
 
+INCLUDES += -I /usr/include/harfbuzz/
 
 #---------------------------------#
 # Definitions of private variables
