Index: src/FTVectoriser.cpp
===================================================================
--- src/FTVectoriser.cpp	(revision 901)
+++ src/FTVectoriser.cpp	(working copy)
@@ -166,7 +166,7 @@
     for(int i = 0; i < ftContourCount; ++i)
     {
         FT_Vector* pointList = &outline.points[startIndex];
-        char* tagList = &outline.tags[startIndex];
+        char* tagList = (char *) &outline.tags[startIndex];
 
         endIndex = outline.contours[i];
         contourLength =  (endIndex - startIndex) + 1;
