Index: src/im_jp2.mak
===================================================================
--- src/im_jp2.mak	(revision 820)
+++ src/im_jp2.mak	(working copy)
@@ -32,6 +32,7 @@
 # NOT necessary if using another distribution
 SRC += jas_binfile.c $(SRCJP2)
 INCLUDES = $(JASPER)
+INCLUDES += ../include/
 DEFINES  = EXCLUDE_JPG_SUPPORT EXCLUDE_MIF_SUPPORT EXCLUDE_PNM_SUPPORT \
            EXCLUDE_BMP_SUPPORT EXCLUDE_PGX_SUPPORT EXCLUDE_RAS_SUPPORT \
            EXCLUDE_TIFF_SUPPORT JAS_GEO_OMIT_PRINTING_CODE JAS_BINFILE $(JASPERVER)
Index: src/im_process.mak
===================================================================
--- src/im_process.mak	(revision 820)
+++ src/im_process.mak	(working copy)
@@ -49,3 +49,5 @@
     endif
   endif
 endif
+
+INCLUDES = . ../include 
Index: src/imlua5.mak
===================================================================
--- src/imlua5.mak	(revision 820)
+++ src/imlua5.mak	(working copy)
@@ -22,6 +22,7 @@
 SRCLUADIR = lua5
 
 INCLUDES = lua5
+INCLUDES = += ../include
 
 ifdef USE_LUA_VERSION
   USE_LUA51:=
Index: src/imlua_fftw35.mak
===================================================================
--- src/imlua_fftw35.mak	(revision 820)
+++ src/imlua_fftw35.mak	(working copy)
@@ -21,6 +21,7 @@
 SRCLUADIR = lua5
 
 INCLUDES = lua5
+INCLUDES = += ../include
 LIBS = im_fftw3
 
 ifdef USE_LUA_VERSION
Index: src/imlua_jp25.mak
===================================================================
--- src/imlua_jp25.mak	(revision 820)
+++ src/imlua_jp25.mak	(working copy)
@@ -12,6 +12,7 @@
 LIBS = im_jp2
 
 INCLUDES = lua5
+INCLUDES = += ../include
 
 ifdef USE_LUA_VERSION
   USE_LUA51:=
Index: src/imlua_process5.mak
===================================================================
--- src/imlua_process5.mak	(revision 820)
+++ src/imlua_process5.mak	(working copy)
@@ -22,6 +22,8 @@
 
 DEFINES += IM_PROCESS
 INCLUDES = lua5
+INCLUDES = += ../include
+
 LIBS = im_process
 
 ifdef USE_LUA_VERSION
Index: src/libjasper2/base/jas_stream.c
===================================================================
--- src/libjasper2/base/jas_stream.c	(revision 820)
+++ src/libjasper2/base/jas_stream.c	(working copy)
@@ -74,6 +74,9 @@
 /* The configuration header file should be included first. */
 #include "jasper/jas_config.h"
 
+// add: kenjirofukuda@gmail.com
+#define JAS_HAVE_UNISTD_H 1
+
 #include <assert.h>
 #if defined(JAS_HAVE_FCNTL_H)
 #include <fcntl.h>
