Index: srccd/config.mak
===================================================================
--- srccd/config.mak	(revision 5971)
+++ srccd/config.mak	(working copy)
@@ -5,11 +5,14 @@
 DEFINES = CD_NO_OLD_INTERFACE
 SRC = iup_cd.c iup_cdutil.c iup_draw_cd.c
 DEF_FILE = iupcd.def
+STDFLAGS = -Wincompatible-pointer-types
 
 USE_CD=YES
 # Can not use USE_IUP because Tecmake will include "iupcd" in linker
 
 INCLUDES = ../include ../src
+INCLUDES += ../../cd/include
+
 LIBS = iup
 LDIR = ../lib/$(TEC_UNAME)
 
Index: srccd/iup_cd.c
===================================================================
--- srccd/iup_cd.c	(revision 5971)
+++ srccd/iup_cd.c	(working copy)
@@ -21,6 +21,7 @@
    So no need to rebuild IUPCD when cdCanvas is changed. */
 #include <cd_private.h>
 
+typedef void (*CanvasVoidPtrArgProc)(cdCanvas* canvas, void* data);
 
 static void (*cdcreatecanvasNATIVE)(cdCanvas* canvas, void* data) = NULL;
 
@@ -82,7 +83,7 @@
   cdcreatecanvasNATIVE = ctx->cxCreateCanvas;
 
   cdIupContext = *ctx;
-  cdIupContext.cxCreateCanvas = cdcreatecanvasIUP;
+  cdIupContext.cxCreateCanvas = (CanvasVoidPtrArgProc) cdcreatecanvasIUP;
 
   return &cdIupContext;
 }
@@ -123,7 +124,7 @@
   cdcreatecanvasDBUFFER = ctx->cxCreateCanvas;
 
   cdIupContextDBuffer = *ctx;
-  cdIupContextDBuffer.cxCreateCanvas = cdcreatecanvasIUP_DBUFFER;
+  cdIupContextDBuffer.cxCreateCanvas = (CanvasVoidPtrArgProc) cdcreatecanvasIUP_DBUFFER;
 
   return &cdIupContextDBuffer;
 }
@@ -164,7 +165,7 @@
   cdcreatecanvasDBUFFERRGB = ctx->cxCreateCanvas;
 
   cdIupContextDBufferRGB = *ctx;
-  cdIupContextDBufferRGB.cxCreateCanvas = cdcreatecanvasIUP_DBUFFERRGB;
+  cdIupContextDBufferRGB.cxCreateCanvas = (CanvasVoidPtrArgProc) cdcreatecanvasIUP_DBUFFERRGB;
 
   return &cdIupContextDBufferRGB;
 }
Index: srccd/iup_draw_cd.c
===================================================================
--- srccd/iup_draw_cd.c	(revision 5971)
+++ srccd/iup_draw_cd.c	(working copy)
@@ -18,6 +18,7 @@
 #include "iup_drvdraw.h"
 #include "iup_drvfont.h"
 
+typedef void (*CanvasVoidPtrArgProc)(cdCanvas* canvas, void* data);
 
 struct _cdCtxCanvas
 {
@@ -520,7 +521,7 @@
                  CD_CAP_LINECAP | CD_CAP_LINEJOIN | CD_CAP_PATH | CD_CAP_BEZIER |
                  CD_CAP_PALETTE | CD_CAP_CLIPPOLY | CD_CAP_REGION | CD_CAP_CHORD),
   CD_CTX_WINDOW,
-  cdcreatecanvas,
+  (CanvasVoidPtrArgProc) cdcreatecanvas,
   cdinittable,
   NULL,
   NULL,
Index: srccontrols/config.mak
===================================================================
--- srccontrols/config.mak	(revision 5971)
+++ srccontrols/config.mak	(working copy)
@@ -3,7 +3,9 @@
 OPT = YES
 
 INCLUDES = ../include ../src . ../srccd
+INCLUDES += ../../cd/include
 
+
 SRCMATRIXEX = iup_matrixex.c iupmatex_clipboard.c iupmatex_busy.c \
               iupmatex_export.c iupmatex_visible.c iupmatex_copy.c \
               iupmatex_units.c iupmatex_find.c iupmatex_undo.c \
Index: srcglcontrols/config.mak
===================================================================
--- srcglcontrols/config.mak	(revision 5971)
+++ srcglcontrols/config.mak	(working copy)
@@ -43,3 +43,6 @@
     BUILD_DYLIB=Yes
   endif
 endif
+
+# add from iup-go
+LIBS += dbus-1
Index: srcim/config.mak
===================================================================
--- srcim/config.mak	(revision 5971)
+++ srcim/config.mak	(working copy)
@@ -9,9 +9,14 @@
 endif  
 
 INCLUDES = ../include ../src
+INCLUDES += ../../im/include
+
 LDIR = ../lib/$(TEC_UNAME)  
 LIBS = iup
 
+# add from iup-go
+LIBS += dbus-1
+
 SRC = iup_im.c
 
 ifneq ($(findstring MacOS, $(TEC_UNAME)), )
Index: srcimglib/config.mak
===================================================================
--- srcimglib/config.mak	(revision 5971)
+++ srcimglib/config.mak	(working copy)
@@ -40,6 +40,9 @@
 LIBS = iup
 LDIR = ../lib/$(TEC_UNAME)
 
+# add from iup-go
+LIBS += dbus-1
+
 ifneq ($(findstring MacOS, $(TEC_UNAME)), )
   ifneq ($(TEC_SYSMINOR), 4)
     BUILD_DYLIB=Yes
Index: srcledc/y.tab.c
===================================================================
--- srcledc/y.tab.c	(revision 5971)
+++ srcledc/y.tab.c	(working copy)
@@ -17,6 +17,7 @@
 #include <errno.h>
 #include "ledc.h"
 
+extern int yylex(void);
 static int yyparse(void);
 
 #line 11 "ledc.y"
@@ -274,7 +275,8 @@
 
     yyssp = yyss;
     yyvsp = yyvs;
-    *yyssp = yystate = 0;
+    *yyssp = 0;
+    yystate = 0;
 
 yyloop:
     if (yyn = yydefred[yystate]) goto yyreduce;
Index: srclua5/config.mak
===================================================================
--- srclua5/config.mak	(revision 5971)
+++ srclua5/config.mak	(working copy)
@@ -84,3 +84,6 @@
 
 SRC = iuplua.c iuplua_api.c iuplua_draw.c iuplua_tree_aux.c iuplua_scanf.c \
       iuplua_getparam.c iuplua_getcolor.c iuplua_config.c $(GC)
+
+# add from iup-go
+LIBS += dbus-1
Index: srclua5/ctrl/iuplua_mglplot.c
===================================================================
--- srclua5/ctrl/iuplua_mglplot.c	(revision 5971)
+++ srclua5/ctrl/iuplua_mglplot.c	(working copy)
@@ -59,7 +59,7 @@
   int count = luaL_checkinteger(L, 6);
   px = iuplua_checkstring_array(L, 4, count);
   py = iuplua_checkdouble_array(L, 5, count);
-  IupMglPlotInsert1D(iuplua_checkihandle(L,1), luaL_checkinteger(L,2), luaL_checkinteger(L,3), px, py, count);
+  IupMglPlotInsert1D(iuplua_checkihandle(L,1), luaL_checkinteger(L,2), luaL_checkinteger(L,3), (const char **)px, py, count);
   free(px);
   free(py);
   return 0;
@@ -98,7 +98,7 @@
   int count = luaL_checkinteger(L, 5);
   px = iuplua_checkstring_array(L, 3, count);
   py = iuplua_checkdouble_array(L, 4, count);
-  IupMglPlotSet1D(iuplua_checkihandle(L,1), luaL_checkinteger(L,2), px, py, count);
+  IupMglPlotSet1D(iuplua_checkihandle(L,1), luaL_checkinteger(L,2), (const char **)px, py, count);
   free(px);
   free(py);
   return 0;
Index: srclua5/ctrl/iuplua_plot.c
===================================================================
--- srclua5/ctrl/iuplua_plot.c	(revision 5971)
+++ srclua5/ctrl/iuplua_plot.c	(working copy)
@@ -332,7 +332,7 @@
   int count = luaL_checkinteger(L, 6);
   px = iuplua_checkstring_array(L, 4, count);
   py = iuplua_checkdouble_array(L, 5, count);
-  IupPlotInsertStrSamples(iuplua_checkihandle(L,1), luaL_checkinteger(L,2), luaL_checkinteger(L,3), px, py, count);
+  IupPlotInsertStrSamples(iuplua_checkihandle(L,1), luaL_checkinteger(L,2), luaL_checkinteger(L,3), (const char **) px, py, count);
   free(px);
   free(py);
   return 0;
@@ -357,7 +357,7 @@
   int count = luaL_checkinteger(L, 5);
   px = iuplua_checkstring_array(L, 3, count);
   py = iuplua_checkdouble_array(L, 4, count);
-  IupPlotAddStrSamples(iuplua_checkihandle(L,1), luaL_checkinteger(L,2), px, py, count);
+  IupPlotAddStrSamples(iuplua_checkihandle(L,1), luaL_checkinteger(L,2), (const char **) px, py, count);
   free(px);
   free(py);
   return 0;
Index: srclua5/iup_mglplot.mak
===================================================================
--- srclua5/iup_mglplot.mak	(revision 5971)
+++ srclua5/iup_mglplot.mak	(working copy)
@@ -16,6 +16,8 @@
 USE_IUPLUA = Yes
 
 INCLUDES = ../srclua5
+INCLUDES += ../include
+
 DEF_FILE = ctrl/iuplua_mglplot.def
 LIBS = iup_mglplot
 
Index: srclua5/iup_plot.mak
===================================================================
--- srclua5/iup_plot.mak	(revision 5971)
+++ srclua5/iup_plot.mak	(working copy)
@@ -17,6 +17,8 @@
 USE_CDLUA = Yes
 
 INCLUDES =  ../src ../srclua5
+INCLUDES += ../include ../../cd/include
+
 LIBS = iup_plot
 DEF_FILE = ctrl/iuplua_plot.def
 
Index: srclua5/iup_scintilla.mak
===================================================================
--- srclua5/iup_scintilla.mak	(revision 5971)
+++ srclua5/iup_scintilla.mak	(working copy)
@@ -16,6 +16,7 @@
 USE_IUPLUA = Yes
 
 INCLUDES = ../srclua5
+INCLUDES += ../src ../include
 DEF_FILE = ctrl/iuplua_scintilla.def
 LIBS = iup_scintilla
 
@@ -73,6 +74,7 @@
 	$(LUABIN) generator.lua $<
 
 SRC := $(GC)
+SRC += ../src/mot/iupunix_notify.c
 
 ifneq ($(findstring MacOS, $(TEC_UNAME)), )
   USE_IUPLUA:=
Index: srclua5/iupcd.mak
===================================================================
--- srclua5/iupcd.mak	(revision 5971)
+++ srclua5/iupcd.mak	(working copy)
@@ -16,6 +16,7 @@
 DEPENDDIR = dep
 
 INCLUDES = ../include
+INCLUDES += ../../cd/include
 LDIR = ../lib/$(TEC_UNAME)
 DEFINES = CD_NO_OLD_INTERFACE
 SRC = iuplua_cd.c
Index: srclua5/iupcontrols.mak
===================================================================
--- srclua5/iupcontrols.mak	(revision 5971)
+++ srclua5/iupcontrols.mak	(working copy)
@@ -17,6 +17,7 @@
 USE_CDLUA = Yes
 
 INCLUDES = ../src ctrl ../srclua5
+INCLUDES +=  ../include ../../cd/include
 LIBS = iupcontrols
 DEF_FILE = ctrl/iupluacontrols.def
 
Index: srclua5/iupgl.mak
===================================================================
--- srclua5/iupgl.mak	(revision 5971)
+++ srclua5/iupgl.mak	(working copy)
@@ -16,6 +16,8 @@
 USE_IUPLUA = Yes
 
 INCLUDES = ../srclua5
+INCLUDES += ../include
+
 DEF_FILE = ctrl/iupluagl.def
 
 ifdef USE_LUA_VERSION
Index: srclua5/iupglcontrols.mak
===================================================================
--- srclua5/iupglcontrols.mak	(revision 5971)
+++ srclua5/iupglcontrols.mak	(working copy)
@@ -16,6 +16,8 @@
 USE_IUPLUA = Yes
 
 INCLUDES = ../src gl ../srclua5
+INCLUDES += ../include
+
 LIBS = iupglcontrols iuplua$(LUASFX)
 DEF_FILE = gl/iupluaglcontrols.def
 
Index: srclua5/iupim.mak
===================================================================
--- srclua5/iupim.mak	(revision 5971)
+++ srclua5/iupim.mak	(working copy)
@@ -17,6 +17,7 @@
 SRC = iuplua_im.c
 DEF_FILE = iupluaim.def
 INCLUDES = ../src
+INCLUDES += ../include ../../im/include
 LIBS = iupim
 
 ifdef USE_LUA_VERSION
@@ -60,3 +61,6 @@
   USE_IMLUA:=
   USE_IM = Yes
 endif
+
+# add from iup-go
+LIBS += dbus-1
Index: srclua5/iupimglib.mak
===================================================================
--- srclua5/iupimglib.mak	(revision 5971)
+++ srclua5/iupimglib.mak	(working copy)
@@ -56,3 +56,8 @@
 ifneq ($(findstring MacOS, $(TEC_UNAME)), )
   USE_IUPLUA:=
 endif
+
+# add from iup-go
+LIBS += dbus-1
+INCLUDES += ../include
+
Index: srclua5/iuplua_api.c
===================================================================
--- srclua5/iuplua_api.c	(revision 5971)
+++ srclua5/iuplua_api.c	(working copy)
@@ -571,7 +571,7 @@
 
   ret = IupListDialog(type, luaL_checkstring(L, 2), 
                             size, 
-                            list, 
+                            (const char **)list, 
                             (int)luaL_checkinteger(L, 5), 
                             (int)luaL_checkinteger(L, 6),
                             (int)luaL_checkinteger(L, 7),
Index: srclua5/iuplua_im.c
===================================================================
--- srclua5/iuplua_im.c	(revision 5971)
+++ srclua5/iuplua_im.c	(working copy)
@@ -83,7 +83,7 @@
 {
   int file_count = (int)luaL_checkinteger(L, 2);
   char** filename_list = iuplua_checkstring_array(L, 1, file_count);
-  Ihandle* animation = IupLoadAnimationFrames(filename_list, file_count);
+  Ihandle* animation = IupLoadAnimationFrames((const char **)filename_list, file_count);
   iuplua_pushihandle(L, animation);
   return 1;
 }
Index: srclua5/iupluascripterdlg.mak
===================================================================
--- srclua5/iupluascripterdlg.mak	(revision 5971)
+++ srclua5/iupluascripterdlg.mak	(working copy)
@@ -16,6 +16,8 @@
 USE_IUPLUA = Yes
 
 INCLUDES = ../src ../srclua5
+INCLUDES += ../include
+
 DEF_FILE = scripter/iupluascripterdlg.def
 LIBS = iup_scintilla iupimglib
 
Index: srclua5/iuptuio.mak
===================================================================
--- srclua5/iuptuio.mak	(revision 5971)
+++ srclua5/iuptuio.mak	(working copy)
@@ -17,6 +17,7 @@
 
 DEF_FILE = ctrl/iupluatuio.def
 INCLUDES = ../src ../srclua5
+INCLUDES += ../include
 LIBS = iuptuio
 
 ifdef USE_LUA_VERSION
Index: srclua5/iupweb.mak
===================================================================
--- srclua5/iupweb.mak	(revision 5971)
+++ srclua5/iupweb.mak	(working copy)
@@ -16,6 +16,8 @@
 USE_IUPLUA = Yes
 
 INCLUDES = ../srclua5
+INCLUDES += ../include
+
 LIBS = iupweb
 DEF_FILE = ctrl/iupluaweb.def
 
Index: srcluaconsole/config.mak
===================================================================
--- srcluaconsole/config.mak	(revision 5971)
+++ srcluaconsole/config.mak	(working copy)
@@ -239,3 +239,5 @@
   OPTFLAGS = -mminimal-toc -ansi -pedantic 
   LFLAGS = -Xlinker "-bbigtoc"
 endif
+
+INCLUDES += ../include
Index: srcmglplot/config.mak
===================================================================
--- srcmglplot/config.mak	(revision 5971)
+++ srcmglplot/config.mak	(working copy)
@@ -32,6 +32,8 @@
 SRCMGLPLOT := $(addprefix src/, $(SRCMGLPLOT))
 
 SRC = iup_mglplot.cpp $(SRCMGLPLOT)
+# add from iup-go
+LIBS += dbus-1
 
 ifneq ($(findstring MacOS, $(TEC_UNAME)), )
   INCLUDES += $(X11_INC)
Index: srcplot/config.mak
===================================================================
--- srcplot/config.mak	(revision 5971)
+++ srcplot/config.mak	(working copy)
@@ -17,6 +17,8 @@
 DEF_FILE = iup_plot.def
 
 INCLUDES = ../include ../src ../srccd
+INCLUDES += ../../cd/include
+
 LDIR = ../lib/$(TEC_UNAME)  
 LIBS = iup iupgl iupcd cdgl 
 
Index: srcscintilla/config.mak
===================================================================
--- srcscintilla/config.mak	(revision 5971)
+++ srcscintilla/config.mak	(working copy)
@@ -269,3 +269,6 @@
     BUILD_DYLIB=Yes
   endif
 endif
+
+# add from iup-go
+LIBS += dbus-1
Index: srcview/config.mak
===================================================================
--- srcview/config.mak	(revision 5971)
+++ srcview/config.mak	(working copy)
@@ -67,3 +67,5 @@
 endif
 
 INCLUDES = ../src
+INCLUDES += ../include
+LIBS += dbus-1
Index: srcview/iup_view.c
===================================================================
--- srcview/iup_view.c	(revision 5971)
+++ srcview/iup_view.c	(working copy)
@@ -202,7 +202,7 @@
   if (!all)
     count = 3;
 
-  ret = IupListDialog(1,"File Format",count,options,1,9,count+1,NULL);
+  ret = IupListDialog(1,"File Format",count,(const char **) options,1,9,count+1,NULL);
   if (ret == -1)
     return NULL;
   else
Index: srcvled/config.mak
===================================================================
--- srcvled/config.mak	(revision 5971)
+++ srcvled/config.mak	(working copy)
@@ -130,3 +130,8 @@
 endif
 
 INCLUDES = ../src
+INCLUDES += ../include ../../cd/include ../../im/include
+
+# add from iup-go
+LIBS += dbus-1
+
Index: srcvled/vled_image_editor.c
===================================================================
--- srcvled/vled_image_editor.c	(revision 5971)
+++ srcvled/vled_image_editor.c	(working copy)
@@ -1723,7 +1723,7 @@
 
     qsort(image_names, num_images, sizeof(Ihandle*), compare_names);
 
-    ret = IupListDialog(1, "Open Image", num_images, image_names, 1, 30, 10, NULL);
+    ret = IupListDialog(1, "Open Image", num_images, (const char **) image_names, 1, 30, 10, NULL);
     if (ret != -1)
     {
       char* name = image_names[ret];
Index: srcweb/config.mak
===================================================================
--- srcweb/config.mak	(revision 5971)
+++ srcweb/config.mak	(working copy)
@@ -3,6 +3,8 @@
 OPT = YES
 
 INCLUDES =  ../include ../src
+INCLUDES += ../include
+
 LDIR = ../lib/$(TEC_UNAME)  
 LIBS = iup
 SRC = iup_webbrowser.c
Index: tecmake.mak
===================================================================
--- tecmake.mak	(revision 5971)
+++ tecmake.mak	(working copy)
@@ -1389,8 +1389,8 @@
   
   ifdef USE_PKGCONFIG
     # get compile/link flags via pkg-config
-    PKGINCS += $(shell pkg-config --cflags gtk+-$(GTKSFX).0 gdk-$(GTKSFX).0 gtk+-unix-print-$(GTKSFX).0)
-    PKGLIBS += $(shell pkg-config --libs gtk+-$(GTKSFX).0 gdk-$(GTKSFX).0 gtk+-unix-print-$(GTKSFX).0)
+    PKGINCS += $(shell pkg-config --cflags gtk+-$(GTKSFX).0 gdk-$(GTKSFX).0 gtk+-unix-print-$(GTKSFX).0 gmodule-2.0)
+    PKGLIBS += $(shell pkg-config --libs gtk+-$(GTKSFX).0 gdk-$(GTKSFX).0 gtk+-unix-print-$(GTKSFX).0 gmodule-2.0)
     GTK_BASE := $(shell pkg-config --variable=prefix gtk+-$(GTKSFX).0)
     GTK := $(GTK_BASE)    
   else
